socketioxide/src/client.rs

```diff
@@ -59,16 +59,17 @@ impl<A: Adapter> Client<A> {
     /// Called when a socket connects to a new namespace
     fn sock_connect(
         &self,
-        auth: Value,
+        auth: Option<Value>,
         ns_path: String,
         socket: &EIoSocket<Self>,
     ) -> Result<(), SendError> {
         debug!("auth: {:?}", auth);
         let handshake = Handshake::new(auth, socket.req_data.clone());
         let sid = socket.sid;
         if let Some(ns) = self.get_ns(&ns_path) {
+            let protocol = socket.protocol;
             let socket = ns.connect(sid, socket.tx.clone(), handshake, self.config.clone());
-            socket.send(Packet::connect(ns_path.clone(), sid))?;
+            socket.send(Packet::connect(ns_path.clone(), sid, protocol))?;
```

Comment:
You need to send the connect response only when it is not a _default ns connection_. I think this is the cause of some of your issues
```suggestion
  if protocol == engineioxide::service::ProtocolVersion::V4 || ns_path != "/" {
      socket.send(Packet::connect(ns_path.clone(), sid, protocol))?;
  }
```